src/main/java/com/mineboundteam/minebound/crafting/serializers/ArmorForgeSerializer.java

```diff
@@ -0,0 +1,54 @@
+package com.mineboundteam.minebound.crafting.serializers;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.mineboundteam.minebound.crafting.ArmorForgeRecipe;
+import net.minecraft.core.NonNullList;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.minecraft.world.item.crafting.RecipeSerializer;
+import net.minecraft.world.item.crafting.ShapedRecipe;
+import net.minecraftforge.registries.ForgeRegistryEntry;
+import org.jetbrains.annotations.NotNull;
+
+public class ArmorForgeSerializer<T> extends ForgeRegistryEntry<RecipeSerializer<?>> implements RecipeSerializer<ArmorForgeRecipe> {
+    @Override
+    public @NotNull ArmorForgeRecipe fromJson(@NotNull ResourceLocation resourceLocation, @NotNull JsonObject jsonObject) {
+        String group = GsonHelper.getAsString(jsonObject, "group", "");
+        JsonArray jsonArray = GsonHelper.getAsJsonArray(jsonObject, "ingredients");
```

Comment:
Please add an additional object called "armor" or something that represents the necessary previous armor slot. Can be null/not present